mocks/aws-sdk-go/elasticache/ElastiCacheAPIOutput.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```

Comment:
A couple things about this that I would suggest changing.

You definitely don't want to place any of this code into the `mocks/` directory. Why? Because of this:

https://github.com/aws-controllers-k8s/elasticache-controller/blob/1313e661427da6c2a886d82d72d7fedb3187dc4a/Makefile#L29-L30

:)

First, I would suggest putting generic test utility code into `pkg/testutil` package. I recommend placing all test fixtures (the .json files) into a `pkg/resource/$RESOURCE/testdata` directory. See [this post by Dave Cheney](https://dave.cheney.net/2016/05/10/test-fixtures-in-go) about why you should call the directory `testdata`.

Secondly, you may not need functions like the `newMockOutput` or `getFileContents` functions if you created a generic fixture loader in `pkg/testutil` that looks something like this:

```go
package testutil

import (
	"encoding/json"
	"io/ioutil"
)

// LoadFromFixture fills an empty pointer variable with the
// data from a fixture JSON file.
func LoadFromFixture(
	fixturePath string,
	output interface{}, // output should be an addressable type (i.e. a pointer)
) {
	contents, err := ioutil.ReadFile(fixturePath)
	if err != nil {
		panic(err)
	}
	err = json.Unmarshal(contents, output)
	if err != nil {
		panic(err)
	}
}
```

You could then call this function from `pkg/resource/replicationgroup/custom_update_api_test.go`:

```go
...
	import "github.com/aws-controllers-k8s/elasticache-controller/pkg/testutil"
...
	var mockOutput svcsdk.DecreaseReplicaCountOutput
	testutil.LoadFromFixture("testdata/DecreaseReplicaCountOutput.json", &mockOutput)
```

Reply:
Thanks for your suggestions Jay. The resultant code is simpler.
I have updated the changes in this PR.